lib/config/providers/chat_provider.dart

```diff
@@ -348,6 +352,40 @@ class ChatNotifier extends Notifier<ChatState> {
     return state.getUnreadCountForGroup(groupId);
   }
 
+  Future<void> refreshUnreadCount(String groupId) async {
+    try {
+      final activePubkey = ref.read(activePubkeyProvider) ?? '';
+      if (activePubkey.isEmpty) {
+        return;
+      }
+
+      final lastRead = await LastReadService.getLastRead(
+        groupId: groupId,
+        activePubkey: activePubkey,
+      );
+
+      final messages = state.groupMessages[groupId] ?? [];
+      final baseline = lastRead ?? DateTime.fromMillisecondsSinceEpoch(0);
+      final unread = messages.where((m) => !m.isMe && m.createdAt.isAfter(baseline)).length;
+
+      state = state.copyWith(
+        unreadCounts: {
+          ...state.unreadCounts,
+          groupId: unread,
+        },
+      );
+    } catch (e, st) {
+      _logger.warning('Failed to refresh unread count for $groupId', e, st);
+    }
+  }
+
+  Future<void> refreshAllUnreadCounts() async {
```

Comment:
Same, would be nice to add tests to new methods if the provider already has a test file (this one has)